pilot/pkg/model/push_context.go

```diff
@@ -793,6 +793,10 @@ func (ps *PushContext) updateContext(
 			authzChanged = true
 		case schemas.AuthenticationPolicy.Type, schemas.AuthenticationMeshPolicy.Type:
 			authnChanged = true
+		case "":
```

Comment:
Can we do something like call the full push context update here or return `(needsFull bool, error)` here and return `(true, nil)`? This is fixing one special case but will make it so any future issues like this will still be really hard to debug and easy to break

Reply:
Yeah I agree. We need to be a bit more careful about the incremental stuff here. Blindly stating that an empty config type change implies virtual service and gateway is not a good idea, esp after @hzxuzhonghu made some recent changes to optimize xds pushes

Reply:
The fix is mainly for quick backporting to 1.4.  

I'm not sure what's the right long-term solution - having the set of booleans and the general structure is far from ideal, and the method looks very scary. Do we have clear benchmarks showing this method really saves a lot to be worth it ?  It seems to still do List in the changed
types. 

In any case - I replaced the fix with generating the type in the controller - and probably for 1.5 
we can clean up a bit the code to generate the full object.

Reply:
Also keep in mind this will also be needed for converting the new K8S Gateway, so a cleanup/rev
of the K8S Ingress support would be great for 1.5.